flink-test-utils-parent/flink-test-utils/src/main/java/org/apache/flink/streaming/util/TestStreamEnvironment.java

```diff
@@ -69,13 +74,28 @@ public static void setAsContext(
                     TestStreamEnvironment env =
                             new TestStreamEnvironment(
                                     miniCluster, parallelism, jarFiles, classpaths);
+                    randomize(conf);
                     env.configure(conf, env.getUserClassloader());
                     return env;
                 };
 
         initializeContextEnvironment(factory);
     }
 
+    /**
+     * Randomizes configuration on test case level even if mini cluster is used in a class rule.
+     *
+     * @param conf
+     */
+    private static void randomize(Configuration conf) {
+        if (Randomization) {
+            final String testName = TestNameProvider.getCurrentTestName();
```

Comment:
hmm...like, we _could_ add a hook to setAsContext to pass the name/modify the config/pass a factory, but it'd still suck since the TestNameProvider must live within the `TestLogger`; we just can't get around that singleton access can we...

Reply:
To make it independent of `TestLogger` we could use [RunListener](https://junit.org/junit4/javadoc/4.12/org/junit/runner/notification/RunListener.html) and then have a singleton listener to fetch the name.
However, I didn't go fully into this direction as it's quite a bit of fiddling in junit4 and I hope to eventually migrate to junit 5. Then I'd check if it can be done in a better way.

Reply:
yeah I'd revisit it once junit5 is in.